### DISCUSSION TITLE
Expiring Automation tokens?

### DISCUSSION THREAD

**Discussion by stagas** [npm/feedback#756] [Registry] [answered]
I am publishing around 30-40 packages in batch once every couple of weeks, I was doing it manually but I found out you can generate an Automation Token that bypasses 2FA for publishing, so my idea is that I want to automate those publishes for that particular day. Is it possible to have an Expiry field when generating the Automation (or other) token so that I can pick, e.g "Expire after 1 day" so that this token (that has full access bypassing 2FA) isn't sitting around anywhere and there's no need to manually go and revoke it (which I could forget or be lazy enough to just let it be the same one, which would be a security risk.). 
Thanks.

**Comment by monishcm** [comment, marked as answer]
Thanks for the feedback. We are working on a similar feature, and you will see more communication around it soon.
